src/app_metadata.c

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ *   Ledger - Secure firmware
+ *   (c) 2023 Ledger
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+
+#if !defined(HAVE_BOLOS)
+
+#include <stdint.h>
+
+#include "bolos_target.h"
+
+#define STR_IMPL_(x) #x
+#define STRINGIFY(x) STR_IMPL_(x)
+
+#define CREATE_METADATA_STRING_ITEM(ITEM_NAME, section_name) \
```

Comment:
Maybe add a do {} while (0) for multiline macro

Reply:
Can't be done as this macro are used for variable definition.